lib/src/isolate_manager/isolate_manager_interface.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:async';
+
+import 'package:dash_kit_network/dash_kit_network.dart';
+
+// This is an abstract class for managing isolates It defines the basic methods
+// needed to start, stop and send messages to an isolate. Flutter Web doesn't
+// support isolates so the `IsolateManager` class has a different implementation
+// for web.
+abstract class IsolateManagerInterface {
+  // `_isolateStatus` keeps track of the current status of the isolate.
+  IsolateStatus _isolateStatus = IsolateStatus.created;
+
+  // Method for starting the isolate. Should be called before sending messages.
+  // If `start()` is called multiple times, an assertion error is thrown.
+  Future<void> start() async {
+    assert(_isolateStatus == IsolateStatus.created, 'Isolate already started');
+    _isolateStatus = IsolateStatus.initializing;
+
+    return;
+  }
+
+  /// Method for stopping the isolate.
+  ///
+  /// Should be called when the isolate is no longer needed.
+  /// Throws an assertion error if the isolate is not started.
+  void stop() {
+    assert(_isolateStatus == IsolateStatus.initialized, 'Isolate not started');
+    _isolateStatus = IsolateStatus.stopped;
+  }
+
+  /// Method for sending messages to the isolate.
+  ///
+  /// The `mapper` function is executed in the isolate and the result is
+  /// returned. Throws an assertion error if the isolate was stopped.
+  /// ```dart
+  ///  IsolateManager isolateManager = IsolateManager();
+  ///  await isolateManager.start();
+  ///  final result = await isolateManager.send(
+  ///   response: response,
+  ///   mapper: responseMapper,
+  ///  );
+  /// ```
+  Future<FutureOr<T>> send<T>({
+    required Response<dynamic> response,
```

Comment:
It is probably will be useful to have some info about the `response` parameter in the doc section for this method

Reply:
Added it in interface class.